projects/netgrif-components-core/src/lib/data-fields/base-component/abstract-base-data-field.component.ts

```diff
@@ -0,0 +1,34 @@
+import {Component, Inject, Input, OnDestroy, Optional} from "@angular/core";
+import {DATA_FIELD_PORTAL_DATA, DataFieldPortalData} from "../models/data-field-portal-data-injection-token";
+import {DataField} from "../models/abstract-data-field";
+import {FormControl} from "@angular/forms";
+import {WrappedBoolean} from "../data-field-template/models/wrapped-boolean";
+
+@Component({
+    selector: 'ncc-base-data-field',
+    template: ''
+})
+export abstract class AbstractBaseDataFieldComponent<T extends DataField<unknown>> implements OnDestroy {
+
+    @Input() public dataField: T;
+    @Input() public formControlRef: FormControl;
+    @Input() public showLargeLayout: WrappedBoolean;
+
+    constructor(@Optional() @Inject(DATA_FIELD_PORTAL_DATA) dataFieldPortalData: DataFieldPortalData<T>) {
+        if (!!dataFieldPortalData) {
+            this.dataField = dataFieldPortalData.dataField;
+            this.formControlRef = dataFieldPortalData.formControlRef;
+            this.showLargeLayout = dataFieldPortalData.showLargeLayout;
+            if (!this.formControlRef) {
```

Comment:
I think in this case it is not guaranteed. It is possible, that dataFieldPortalData contains defined formControlRef that is provided through `AbstractDataFieldTemplateComponent` and `AbstractFieldComponentResolver` via datafield object.
But you are right it would be better to check whether datafield has registered formcontrol, I will change it.